1Password/shell-plugins
# update(openai): Support "oaieval" and "oaievalset" as executables within the "openai" shell plugin

This PR introduces a Shell Plugin for [OpenAI's new Evals framework](https://github.com/openai/evals) that was announced as part of the [GPT-4 language model](https://openai.com/product/gpt-4).

- GPT-4 homepage: https://openai.com/product/gpt-4
- Evals framework homepage: https://github.com/openai/evals

This is a fairly straightforward shell plugin as the Evals CLI expects a `OPENAI_API_KEY` environment variable with the OpenAI API Key which can be obtained on the [OpenAI API Keys page](https://platform.openai.com/account/api-keys). There's no CLI flag that accepts the API Key, nor a configuration file.

There's a way to set the API Key within an individual Evals module code though, using `openai.api_key` or a path to the configuration file using `openai.api_key_path` but neither are applicable to Shell Plugins because that's within the context of the Evals module code:

```
No API key provided. You can set your API key in code using 'openai.api_key = <API-KEY>', or you can set the environment variable OPENAI_API_KEY=<API-KEY>). If your API key is stored in a file, you can point the openai module at it with 'openai.api_key_path = <PATH>'. You can generate API keys in the OpenAI web interface. See https://onboard.openai.com for details, or email support@openai.com if you have any questions.
```

## Here's how to test

- Clone the [Evals repository](https://github.com/openai/evals) and set it up as [outlined under `Making evals` section](https://github.com/openai/evals#making-evals).
- Checkout the current PR branch and run `make evals/build`
- Set up API Key on 1Password by running `op plugin init oaieval` (API Key can be obtained from [this page](https://platform.openai.com/account/api-keys))
- Future runs for `oaieval` will use the API Key stored on 1Password.
- To test, run an example eval like `oaieval gpt-3.5-turbo test-match` and ensure it runs successfully.

## Blockers

- The Evals framework project introduces not only `oaieval` CLI command but also `oaievalset` to test eval sets.
- As [my understanding stands (from the flyctl PR)](https://github.com/1Password/shell-plugins/pull/141#issuecomment-1469446169), there isn't currently a way to introduce more than one executable command for a shell plugin without changes to the op CLI code. So, I am not sure how to proceed here.
